performance_regression_test.py

```diff
@@ -797,3 +798,115 @@ def build_histogram(self, workload: PerformanceTestWorkload, test_type: Performa
 
         self.update_hdrhistograms(histogram_name='test_histogram_by_interval',
                                   histogram_data=histogram_data_by_interval)
+
+
+class PerformanceRegressionUpgradeTest(PerformanceRegressionTest, UpgradeTest):  # pylint: disable=too-many-ancestors
+    def __int__(self, *args):
+        super().__init__(*args)
+        self.cluster = None  # pylint: disable=attribute-defined-outside-init
+        self.tester = None  # pylint: disable=attribute-defined-outside-init
+        self.monitoring_set = None  # pylint: disable=attribute-defined-outside-init
+
+    def get_email_data(self):  # pylint: disable=no-self-use
+        """prepare data to generate and send via email
```

Comment:
please cange the comment here;

Here you force to use PerformanceRegressionTest 
email tade generation  mechanism instead of UpgradeTest

or maybe even better to call PerformanceRegressionTest.get_email_data function directly here
```
def get_email_data(self):
   return PerformanceRegressionTest.get_email_data(self)
``

Reply:
because of the multiple inheritance, we got into a situation where `get_email_data()` is not implemented in `PerformanceRegressionTest`, but only on `UpgradeTest` leading to errors...
so, in this case, we need to do the same as `PerformanceRegressionTest` does (from its base class) that is `return {}`

Reply:
just look at the code :
```
 Required field:
        - email_template: path to file with html template
        - email_subject: subject of email
        have to be implemented in child class.
```
It looks like an instruction that we need to implement this method

but as you mentioned for propperf work it must call the same method as in  PerformanceRegressionTest 

so we can call the method PerformanceRegressionTest directly to skip MRO  
```
def get_email_data(self):
   return PerformanceRegressionTest.get_email_data(self)
```